src/platform/startup_notify.rs

```diff
@@ -33,6 +34,28 @@ const X11_VAR: &str = "DESKTOP_STARTUP_ID";
 /// The variable which is used mostly on Wayland.
 const WAYLAND_VAR: &str = "XDG_ACTIVATION_TOKEN";
 
+/// The error type for when the requested operation is not supported by the backend.
+#[derive(Debug)]
+pub struct ActivationTokenNotFound {
+    _marker: (),
+}
+
+impl fmt::Display for ActivationTokenNotFound {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "the requested operation is not supported by Winit")
```

Comment:
Use `f.write_str` here for the same reason

Reply:
This is being tracked upstream in https://github.com/rust-lang/rust/issues/99012, and I believe that's the better way to go (e.g. write the "best" Rust code you can, and let the compiler figure out the details (within limits of course)).

Besides, the compilation performance impact here is negligible compared to so many other things in this crate.